shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -497,18 +496,11 @@ fun NearbyStaticData.withRealtimeInfo(
 
     val allDataLoaded = schedules != null
 
-    // this needs to change how the trip keys are constructed
-    val (rewrittenThis, rewrittenPredictions) =
-        if (globalData == null || schedules == null) Pair(this, predictions)
+    val rewrittenThis =
+        if (globalData == null || schedules == null) this
         else
-            TemporaryTerminalRewriter(
-                    this,
-                    predictions,
-                    globalData,
-                    activeRelevantAlerts,
-                    schedules
-                )
-                .rewritten()
+            TemporaryTerminalFilter(this, predictions, globalData, activeRelevantAlerts, schedules)
+                .filtered()
```

Comment:
👍 I like the way you still kept this isolated so it will be easy to remove in the future once the upstream change is made!